api/types/types.go

```diff
@@ -556,6 +556,21 @@ type DiskUsageOptions struct {
 	Types []DiskUsageObject
 }
 
+// ContainerUsage contains container usage part of response of Engine API:
+// GET "/system/df"
+type ContainerUsage struct {
+	ID         string `json:"Id"`
+	Image      string
+	ImageID    string
+	Command    string
+	Created    int64
+	Names      []string
+	SizeRw     int64 `json:",omitempty"`
+	SizeRootFs int64 `json:",omitempty"`
+	Status     string
+	Mounts     []MountPoint
```

Comment:
We probably can't remove this field if we want it to be backward compatible, but looking at how it's used, it was a bit of a weird choice to return all mount information, because the CLI only does a `count` of volumes that use the `local` driver;

```json
{
      "Mounts": [
        {
          "Type": "volume",
          "Name": "some-volume",
          "Source": "/var/lib/docker/volumes/some-volume/_data",
          "Destination": "/foo",
          "Driver": "local",
          "Mode": "z",
          "RW": true,
          "Propagation": ""
        }
      ]
}
```

CLI code does this; https://github.com/docker/cli/blob/a32cd16160f1b41c1c4ae7bee4dac929d1484e59/cli/command/formatter/container.go#L270-L280

```go
// LocalVolumes returns the number of volumes using the "local" volume driver.
func (c *ContainerContext) LocalVolumes() string {
	count := 0
	for _, m := range c.c.Mounts {
		if m.Driver == "local" {
			count++
		}
	}

	return fmt.Sprintf("%d", count)
}
```

Thinking if it would be worth deprecating the field for API >= 1.42, and instead adding a `LocalVolumes` field (or `VolumeCount`) and do the above logic on the daemon side 🤔 🤔 🤔 🤔 
If we want to do that, we could make it a pointer, and `omitempty`, and clear the field on < 1.42

```suggestion
	Mounts     *[]MountPoint  `json:",omitempty"`
```

Reply:
Sounds like a good idea, but let's leave that for a separate PR and focus just on "trimming down" here without changing the existing functionality too much